h2/src/main/org/h2/server/web/WebThread.java

```diff
@@ -250,7 +250,17 @@ private boolean checkHost(String host) throws IOException {
             writeSimple("HTTP/1.1 400 Bad Request", "Bad request");
             return false;
         }
-        int index = host.indexOf(':');
+        int index ;
+
+        if(host.contains("["))
+        {
+            index = host.indexOf(']') +1;
+        }
+        else
+        {
+            index = host.indexOf(':');
+        }
+
```

Comment:
Thank you for your contribution!

Just use `lastIndexOf(':')`, all these complications aren't really needed.

You also need to add `"[::1]"` to the list of unconditionally allowed names below (don't forget to wrap a long line, we don't want lines longer than 120 characters in our source code).